openexr-sys/c_wrapper/rust_ostream.cpp

```diff
@@ -0,0 +1,25 @@
+#include "rust_ostream.hpp"
+
+#include <stdexcept>
+
+using namespace IMATH_NAMESPACE;
+
+void RustOStream::write(const char c[], int n) {
+    int res = write_ptr(writer, c, n);
+    if (res != 0) {
+        throw std::runtime_error("error writing data");
```

Comment:
Ah, thanks for the tip!  Done, to an extent.  It would be cool to figure out a way to turn it back into a proper Rust error (somehow round-trip it through the C++ API) but this seems to work well for now.